test/Chainweb/Test/Pact/Utils.hs

```diff
@@ -80,13 +120,39 @@ getByteString = fst . B16.decode
 formatB16PubKey :: SomeKeyPair -> Text
 formatB16PubKey = toB16Text . formatPublicKey
 
+-- | Merge a list of JSON Objects together. This will defer to 'Null' if any other values
+-- are found. Beware.
+mergeObjects :: [Value] -> Value
+mergeObjects = Object . HM.unions . foldr unwrap []
+  where
+    unwrap (Object o) = (:) o
+    unwrap _ = id
+
+-- | Lift a Maybe into a singleton List
+singletonOf :: Maybe a -> [a]
+singletonOf = toList
+
 mkPactTestTransactions :: Vector String -> IO (Vector ChainwebTransaction)
 mkPactTestTransactions cmdStrs = do
     kps <- testKeyPairs
     let theData = object ["test-admin-keyset" .= fmap formatB16PubKey kps]
     -- using 1 as the nonce here so the hashes match for the same commands (for testing only)
     traverse (mkPactTransaction kps theData "1") cmdStrs
 
+mkPactTestTransactions'
+    :: Vector PactTransaction
+    -> IO (Vector ChainwebTransaction)
+mkPactTestTransactions' txs =
+    testKeyPairs >>= \ks -> traverse (go ks) txs
+  where
+    -- merge tx data and create pact command
+    go ks (PactTransaction c d) =
+      let pd = mergeObjects $ [keys ks] <> singletonOf d
+      in mkPactTransaction ks pd "1" (unpack c)
+
+    -- create public test admin keys from test keyset
+    keys ks = object ["test-admin-keyset" .= fmap formatB16PubKey ks]
+
```

Comment:
This guy now allows us to write transactions _and_ custom data. See the SPV tests for usage